MapboxNavigation/NavigationMapView.swift

```diff
@@ -690,6 +690,45 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         ].compactMap { style.source(withIdentifier: $0) }))
     }
     
+    /**
+     Removes completed sections of the route line.
+     */
+    
+    public func fadeRouteLine(percentComplete: Double) {
+        guard let style = self.style else { return }
+        
+        /**
+         This is necessary to prevent duplicate keys in the dictionary
+         defining the gradient. It's not for aesthetic purposes.
+         */
+        let gradientToTransparent: Double = 0.0005
```

Comment:
https://github.com/mapbox/mapbox-navigation-ios/issues/1307#issuecomment-635461995 discusses what it’ll take to eliminate this fudge factor, which doesn’t work if the route is over a certain length.

If for some reason we need to keep a fudge factor, we might be able to refine it a bit, considering that mbgl considers these stops to be floats. Instead of this constant, cast `percentComplete` to `Float` and add `.nextUp` to it. That property is the smallest possible larger value in the type.